calculator.rb

```diff
@@ -0,0 +1,174 @@
+# Return the output and the string form of the operator
+def calculate(operation, first_num, second_num)	
+
+	first_number_data = check_num_format(first_num)
+	second_number_data = check_num_format(second_num)
+	
+	if $supported_operations[:addition].include?(operation)
+		output = first_number_data + second_number_data
+		operator = $supported_operations[:addition][-1]
+
+	elsif $supported_operations[:subtraction].include?(operation)
+	 	output = first_number_data - second_number_data
+	 	operator = $supported_operations[:subtraction][-1]
+
+	elsif $supported_operations[:multiplication].include?(operation)
+	 	output = first_number_data * second_number_data
+	 	operator = $supported_operations[:multiplication][-1]
+
+	elsif $supported_operations[:division].include?(operation)
+	 	output = first_number_data.to_f / second_number_data.to_f
+	 	operator = $supported_operations[:division][-1]
+
+	elsif $supported_operations[:exponent].include?(operation)
+	 	output = first_number_data ** second_number_data
+	 	operator = $supported_operations[:exponent][-1]
+
+	elsif $supported_operations[:modulus].include?(operation)
+	 	output = first_number_data % second_number_data
+	 	operator = $supported_operations[:modulus][-1]
+	end
+
+	return output, operator
+end
+
+# Evaluate a statement contained in parentheses and return the value
+def evaluate_parens(statement_in_parens)
+	# strip the parens out of the input
+	statement = statement_in_parens[1...-1]
+	
+	# partition the statement using a regex to find the operation
+	# feed the elements of the array into the 2 numbers and operation
+	statement_first_num = statement.partition(/[\+\-\^\*\/\%]/)[0].strip
+	statement_operation = statement.partition(/[\+\-\^\*\/\%]/)[1].strip
+	statement_second_num = statement.partition(/[\+\-\^\*\/\%]/)[2].strip	
+
+	# run the calculate function and save the returned values into variables
+	value, operator = calculate(statement_operation, statement_first_num, statement_second_num)
+	
+	# return just the output value
+	return value
+end
+
+# Ask for what operation the user wants to perform and store the input in a variable.
+puts "What operation would you like to perform?"
+operation = gets.chomp.downcase
+
+# Create a boolean variable to store whether the operation is valid or not.
+valid = false
+
+# Check for operation validity
+while valid == false
+	# If operation falls into the flattened supported operations array, stop the loop and proceed.
+	if $supported_operations.flatten(2).include?(operation)
+		valid = true
+	# If the user pressed enter without entering data, ask them for another operation.
+	elsif operation == ""
+		puts "Please enter an operation."
+		operation = gets.chomp.downcase
+	# If the user entered a word that is not on the supported operations list, ask for another operation.
+	else
+		puts "I do not recognize the operation \"#{operation}\". Please enter: addition, subtraction, multiplication, division, exponent, or modulus."
```

Comment:
It's good that you have catch all messaging to alert the user they're not entering a valid operation and the why. :+1: 

I did notice that users aren't provided with guidance when they're first entering an operator, or if they press enter without entering anything though.